File: WorldFeed/Client/src/app/core/Core/Models/TwitterEntities/UserMentionEntity.ts
import {IUserMentionEntity} from "../../../Public/Models/Entities/IUserMentionEntity";

// Object storing information related with an user mention on Twitter
export class UserMentionEntity implements IUserMentionEntity {
  // [JsonProperty("id")]
  public id: number; // long?

  // [JsonProperty("id_str")]
  public idStr: string;

  // [JsonProperty("screen_name")]
  public screenName: string;

  // [JsonProperty("name")]
  public name: string;

  // [JsonProperty("indices")]
  public indices: Array<number>;

  public equals(other: IUserMentionEntity): boolean {
    if (other == null || !this.id || !other.id) {
      return false;
    }

    if (this.id !== other.id) {
      return false;
    }

    if (this.indices == null || other.indices == null) {
      return this.equals(Indices, other.indices);
    }

    return this.indices.ContainsSameObjectsAs(other.indices, true);
  }

  public ToString(): string {
    return `@${(this.screenName)}`;
  }
}

File: Sharebook/Server/src/Sharebook.Data/Sharebook.Data.Models/Post/Entities/UserMentionEntity.cs
namespace Sharebook.Data.Models.Post.Entities
{
    using global::System.Collections.Generic;
    using Sharebook.Data.Common.Models;

    /// <summary>
    /// Object storing information related with an user mention on Twitter
    /// </summary>
    public class UserMentionEntity : BaseDeletableModel<long>
    {
        public string IdStr { get; set; }

        public string ScreenName { get; set; }

        public string Name { get; set; }

        public int IndicesId { get; set; }
        public Indices Indices { get; set; }

        //public bool Equals(IUserMentionEntity other)
        //{
        //    if (other == null || !Id.HasValue || !other.Id.HasValue)
        //    {
        //        return false;
        //    }

        //    if (Id.Value != other.Id.Value)
        //    {
        //        return false;
        //    }

        //    if (Indices == null || other.Indices == null)
        //    {
        //        return Equals(Indices, other.Indices);
        //    }

        //    return Indices.ContainsSameObjectsAs(other.Indices, true);
        //}

        //public override string ToString()
        //{
        //    return $"@{ScreenName}";
        //}
    }
}
